src/PrefixTree.java

```diff
@@ -13,7 +14,47 @@ public class PrefixTree {
     private static class TraverseResult {
         private Node node;
         private int index;
+    }
+
+    private static class Node {
+
+        private boolean endOfWord;
+        private final char value;
+        private ArrayList<Node> edges = new ArrayList<>();
+
+        Node(char value) {
+            this.value = value;
+        }
+
+        public boolean isEndOfWord() {
+            return endOfWord;
+        }
+
+        public void setEndOfWord(boolean endOfWord) {
+            this.endOfWord = endOfWord;
+        }
```

Comment:
Given that there's no remove operation, this can lose the parameter and become an idempotent way of marking a node as final.